tests/lib/linter.js

```diff
@@ -115,6 +115,24 @@ describe("Linter", () => {
 
             linter.verify("foo + bar", { rules: { checker: "error" } });
         });
+
+        it("does not call rule listeners with a `this` value", () => {
+            const spy = sandbox.spy(function() {
+                assert.strictEqual(this, void 0); // eslint-disable-line no-invalid-this
+            });
+
+            linter.defineRule("checker", () => ({ Program: spy }));
+            linter.verify("foo", { rules: { checker: "error" } });
+            assert(spy.calledOnce);
```

Comment:
I think this would be more readable by adding `assert.notOk(spy.firstCall.thisValue)` so all of the assertions are in one place. But I won't insist on it.